janitor/functions.py

```diff
@@ -96,7 +97,9 @@ def clean_names(
     :param remove_special: (optional) Remove special characters from columns.
         Only letters, numbers and underscores are preserved.
     :returns: A pandas DataFrame.
+    :param original_columns: (optional) Preserve original column names.
     """
+    mylist = list(df.columns)
```

Comment:
On matters of code style, I think `my_list` is not a sufficiently informative name. It's a matter of good Python coding practice to provide variable names that are descriptive. Could you change this line please?

Here's a suggested change:

```suggestion
    # Store the original column names, so that they can be retrieved later on.
    original_column_names = list(df.columns)
```